### ISSUE TITLE
Area2D gravity affects bodies which are no longer in the area

### ISSUE THREAD

**Issue by radishes** [godotengine/godot#77449]
### Godot version

4.1.dev2

### System information

Windows 10, Compatibility renderer

### Issue description

I have an Area2D which follows the cursor. When I click, the Area2D enables point gravity with a negative value using the following code, and also starts a timer for 0.2 seconds. Once the timer expires, the point gravity on the Area2D is disabled. My goal is to create an "anti-gravity burst" which pushes nearby physics objects away from the cursor when the mouse is clicked.

```
$GravityArea2D.gravity_space_override = Area2D.SPACE_OVERRIDE_REPLACE
$GravityArea2D.gravity_point = true
$GravityArea2D.gravity_point_center = Vector2()
$GravityArea2D.gravity = -2000
```

This works as expected, except I notice that sometimes physics objects continue to be influenced by the altered gravity on subsequent clicks, even though the Area2D no longer overlaps the object in question. Because the way this happens is inconsistent, I think it's a bug.

The attached video demonstrates the issue. The larger circle is the Area2D which follows the cursor. The smaller circle is a RigidBody2D ball which can be pushed away using the anti-gravity burst. At Click 27 I begin to influence the ball by clicking near it. Around Click 40 I begin to click away from the ball again, and the ball is not affected because it's not within the Area2D (as expected.)  By the mid-40s I click near the ball again, and it remains influenced by subsequent clicks for the rest of the video even when the Area2D is not overlapping.

### Steps to reproduce

In the repro project: Click near the bouncing ball, then click away from it and see if the ball is still influenced by the Area2D's gravity. Not every click near the ball will cause the ball to gain the lasting influence of the Area2D. Try clicking near the ball many times and then clicking away from it.

### Minimal reproduction project

https://github.com/godotengine/godot/assets/1106565/88301e8d-7c52-49b3-8a81-e5cdc33e0998

[gravity-repro.zip](https://github.com/godotengine/godot/files/11557624/gravity-repro.zip)

**Comment by krazy-j** [comment]
I have also noticed this exact issue in my game. I have an Area2D (stationary and always monitoring) with a space override for gravity. It works as expected most of the time, but sometimes, after a RigidBody2D exits the Area2D, it continues to be influenced by the Area2D's gravity. The RigidBody2D is then permanently affected by the Area2D (until the scene is reloaded).

Some other details I've noticed:
- It seems more likely to occur when the RigidBody2D is moving faster, or when the Area2D is also moving.
- The Area2D correctly reports having no overlapping bodies, despite the RigidBody2D still being influenced by its gravity.
- If I change the gravity direction/magnitude of the Area2D, it affects the RigidBody2D accordingly.
- The RigidBody2D will stop being affected if the Area2D's space override is disabled, but immediately resumes if it's re-enabled.
- Removing the collision shape from the Area2D does not stop the effect.
- Removing the Area2D from world space (via `PhysicsServer2D.area_set_space(get_rid(), RID())`) does not stop the effect.
- Setting `monitoring` to false does not change anything.

I'm hoping this gets fixed, as it's a very significant issue in my game.

**Comment by NolanDC** [comment]
I believe I have reproduced the issue with an extremely simple example. Here is a gist containing a scene that can reproduce this every time: https://gist.github.com/NolanDC/07044dae2c27f3e4e490ececbc88db45. You'll notice the ball just flies upwards into eternity.

I've attached a .zip of the same files as well: [goodbye_ball.zip](https://github.com/godotengine/godot/files/12553024/goodbye_ball.zip)

I've got a fix for this in progress, will have a PR up shortly. But figured it best to add my example here with the others.
